proposed/2021/MachineReadableOutput.md

```diff
@@ -0,0 +1,792 @@
+    "MyProjectA": [
+      {
+        "Path": "src/tool/MyProjectA.csproj",
+        "framework": "netcoreapp3.1",
+        "topLevelPackages": [
+          {
+            "id": "Microsoft.Extensions.Primitives",
+            "requestedVersion": "[1.0.0, 5.0.0]",
+            "resolvedVersion": "1.0.0"
+          },
+          {
+            "id": "NuGet.Commands",
+            "requestedVersion": "4.8.0-preview3.5278",
+            "resolvedVersion": "4.8.0-preview3.5278"
+          },
+          {
+            "id": "Text2Xml.Lib",
+            "requestedVersion": "[1.1.2, 2.0.0)",
+            "resolvedVersion": "1.1.2"
+          }
+        ]
+      }
+    ],
+    "MyProjectB": [
+      {
+        "Path": "src/lib/MyProjectB.csproj",
+        "framework": "netcoreapp3.1",
+        "topLevelPackages": [
+          {
+            "id": "NuGet.Commands",
+            "requestedVersion": "4.8.0-preview3.5278",
+            "resolvedVersion": "4.8.0-preview3.5278"
+          },
+          {
+            "id": "Text2Xml.Lib",
+            "requestedVersion": "1.1.2",
+            "resolvedVersion": "1.1.2"
+          }
+        ]
+      },
+      {
+        "Path": "src/lib/MyProjectB.csproj",
+        "framework": "net5.0",
+        "topLevelPackages": [ 
+          {
+            "id": "NuGet.Commands",
+            "requestedVersion": "4.8.0-preview3.5278",
+            "resolvedVersion": "4.8.0-preview3.5278"
+          },
+          {
+            "id": "Text2Xml.Lib",
+            "requestedVersion": "1.1.2",
+            "resolvedVersion": "1.1.2"
+          }
+        ]
+      }
+    ]
+  }
+}
+```
+
+#### `> dotnet list package --outdated`
+
+```dotnetcli
+
+The following sources were used:
+   https://api.nuget.org/v3/index.json
+   https://apidev.nugettest.org/v3-index/index.json
+
+Project `MyProjectA` has the following updates to its packages
+   [netcoreapp3.1]:
+   Top-level Package                      Requested             Resolved              Latest
+   > Microsoft.Extensions.Primitives      [1.0.0, 5.0.0]        1.0.0                 6.0.0
+   > NuGet.Commands                       4.8.0-preview3.5278   4.8.0-preview3.5278   6.0.0
+   > Text2Xml.Lib                         [1.1.2, 2.0.0)        1.1.2                 1.1.4
+
+Project `MyProjectB` has the following updates to its packages
+   [netcoreapp3.1]:
+   Top-level Package      Requested             Resolved              Latest
+   > NuGet.Commands       4.8.0-preview3.5278   4.8.0-preview3.5278   6.0.0
+   > Text2Xml.Lib         1.1.2                 1.1.2                 1.1.4
+
+   [net5.0]:
+   Top-level Package      Requested             Resolved              Latest
+   > NuGet.Commands       4.8.0-preview3.5278   4.8.0-preview3.5278   6.0.0
+   > Text2Xml.Lib         1.1.2                 1.1.2                 1.1.4
+```
+
+#### `> dotnet list package --outdated --format json`
+
+```json
+{
+  "version": 1,
+  "parameters": "--outdated",
+   "sources": [
```

Comment:
in several of these JSON examples I see the sources listed - how does the Package Source Mapping feature interact with these listings?

Reply:
Without `Package Source Mapping` is not enabled then the package may come from any of those sources, but if it's enabled it only downloaded from specified source. Yes, you're right we may need to include that filtering info to be more precise. 
I created follow up issue https://github.com/NuGet/Home/issues/11557
I don't believe we'll include anything about package source mapping for this iteration.
Another overlapping idea is: https://github.com/NuGet/Home/issues/11552